### PR TITLE
refs #44 blog page addition

### PR DESCRIPTION
**Pull request by emullaney**
Addition of blog page and blog index page models and templates. Blog Index template will need to be updated later along with the marketplace index and case study index when that part is complete.

### CODE REVIEW

File: portal_pages/models.py [hunk 1]
```diff
@@ -428,3 +433,103 @@ class OrganizationCaseStudy(Orderable, models.Model):
...
+    search_fields = Page.search_fields + (
+        index.SearchField('body'),
+    )
+
+    author = models.ForeignKey(
+        RapidProUser,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL
+    )
```
Comment (kmtracey) [inline portal_pages/models.py]: Ultimately its wanted to have (anonymous) users submit blog entries, so I'm not sure a Foreign Key to User makes the most sense for Author field...not all blog authors will have logins on site (the only people with logins envisioned right now are the admins)...

File: portal_pages/models.py [hunk 2]
```diff
@@ -428,3 +433,95 @@ class OrganizationCaseStudy(Orderable, models.Model):
...
+
+        # Filter by tag
+        tag = request.GET.get('tag')
+        if tag:
+            blogs = blogs.filter(tags__name=tag)
+
+        # Filter by author
+        author = request.GET.get('author')
+        if author:
+            blogs = blogs.filter(author__username=author)
```
Comment (kmtracey) [inline portal_pages/models.py]: This needs to be removed for now at least...

File: portal_pages/templates/portal_pages/blog_index_page.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        {% if request.GET.tag|length %}
+            <h4>Showing posts tagged as {{ request.GET.tag|safe }}</h4>
+            <div class="tags"><a href=" {% pageurl self %} ">Show all</a></div>
+        {% endif %}
+
+        {% if request.GET.author|length %}
+            <h4>Showing posts tagged with author {{ request.GET.author|safe }}</h4>
+            <div class="tags"><a href=" {% pageurl self %} ">Show all</a></div>
+        {% endif %}
```
Comment (kmtracey) [inline portal_pages/templates/portal_pages/blog_index_page.html]: This block should be removed.

**Comment by emullaney** [comment]
I've made the recommended changes so this should be good to go, but if you want to give it another pass through before I merge, @kmtracey let me know.

File: portal_pages/models.py [hunk 4]
```diff
@@ -428,3 +433,90 @@ class OrganizationCaseStudy(Orderable, models.Model):
     panels = [
         FieldPanel('organization'),
     ]
+
+
+# Blog index page
+
+
+class BlogIndexPage(Page):
```
Comment (kmtracey) [inline portal_pages/models.py]: Needs to inherit from TopImage I think?

Comment (emullaney) [inline portal_pages/models.py]: Looks like TopImage is already in this branch, not sure which version this was...

**Comment by emullaney** [comment]
@kmtracey I think this is good to merge.

**Comment by kmtracey** [comment]
:+1:

**Pull request merged**